core/src/main/java/com/datastrato/gravitino/meta/RoleEntity.java

```diff
@@ -155,6 +162,23 @@ public Namespace namespace() {
     return namespace;
   }
 
+  public boolean hasPrivilegeWithCondition(
```

Comment:
I would suggest not put privilege checking logic in this object class, you'd create a new method class to implement such logic.

Reply:
I moved the logic to AuthorizationUtils.